From 446480740deb144f5c2e3426c9668db3b031b985 Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 30 Mar 2022 13:28:41 +0300
Subject: [PATCH 049/103] pcie: Remove 'no-check-serdes' dts property

Since current version of S32G3 SoC's support reading from
SIUL2* registers, we can drop the 'no-check-serdes'
workaround support for these platforms.

Issue: ALB-8727
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32g3xxaevb.dts |  8 --------
 drivers/pci/pcie_s32gen1.c       | 18 +++++++-----------
 drivers/pci/pcie_s32gen1.h       |  3 +--
 3 files changed, 8 insertions(+), 21 deletions(-)

diff --git a/arch/arm/dts/fsl-s32g3xxaevb.dts b/arch/arm/dts/fsl-s32g3xxaevb.dts
index 587d831818..b81ffcef9c 100644
--- a/arch/arm/dts/fsl-s32g3xxaevb.dts
+++ b/arch/arm/dts/fsl-s32g3xxaevb.dts
@@ -11,11 +11,3 @@
 / {
 	model = "NXP S32G3XXX-EVB";
 };
-
-&pcie0 {
-	no-check-serdes;
-};
-
-&pcie1 {
-	no-check-serdes;
-};
diff --git a/drivers/pci/pcie_s32gen1.c b/drivers/pci/pcie_s32gen1.c
index 4ab0c1c3cd..2ed07d0060 100644
--- a/drivers/pci/pcie_s32gen1.c
+++ b/drivers/pci/pcie_s32gen1.c
@@ -5,18 +5,18 @@
  */
 
 #include <common.h>
-#include <pci.h>
-#include <asm/io.h>
+#include <dm.h>
 #include <errno.h>
+#include <hwconfig.h>
 #include <malloc.h>
-#include <dm.h>
+#include <misc.h>
+#include <pci.h>
+#include <s32gen1_siul2_nvram.h>
+#include <asm/io.h>
 #include <asm/arch/clock.h>
-#include <linux/sizes.h>
 #include <dm/device-internal.h>
-#include <hwconfig.h>
 #include <dm/uclass.h>
-#include <misc.h>
-#include <s32gen1_siul2_nvram.h>
+#include <linux/sizes.h>
 
 /* CFG1 is used in linux when finding devices on the bus.
  * It is actually the upper half of the config space
@@ -756,9 +756,6 @@ static int s32_pcie_get_config_from_device_tree(struct s32_pcie *pcie)
 	/* get supported width (X1/X2) from device tree */
 	pcie->linkwidth = fdtdec_get_int(fdt, node, "num-lanes", X1);
 
-	if (fdt_getprop(fdt, node, "no-check-serdes", NULL))
-		pcie->no_check_serdes = true;
-
 	return ret;
 }
 
@@ -900,7 +897,6 @@ static int s32_pcie_probe(struct udevice *dev)
 	struct udevice *siul21_nvmem = NULL;
 
 	pcie->enabled = false;
-	pcie->no_check_serdes = false;
 
 	ret = uclass_get_device_by_name(UCLASS_MISC, "siul2_1_nvram",
 					&siul21_nvmem);
diff --git a/drivers/pci/pcie_s32gen1.h b/drivers/pci/pcie_s32gen1.h
index 0df2aee250..e673cab83e 100644
--- a/drivers/pci/pcie_s32gen1.h
+++ b/drivers/pci/pcie_s32gen1.h
@@ -6,8 +6,8 @@
 
 #ifndef PCIE_S32GEN1_H
 #define PCIE_S32GEN1_H
-#include <pci.h>
 #include <dm.h>
+#include <pci.h>
 #include <asm/io.h>
 
 #include "serdes_s32gen1.h"
@@ -152,7 +152,6 @@ struct s32_pcie {
 	int id;
 	bool enabled;
 	bool ep_mode;
-	bool no_check_serdes;
 	enum serdes_link_width linkwidth;
 	enum pcie_link_speed linkspeed;
 	int atu_out_num;
-- 
2.25.1

